typescript/sdk/src/providers/transactions/submitter/TxSubmitter.ts

```diff
@@ -0,0 +1,30 @@
+import { ChainName } from '../../../types.js';
+import { MultiProvider } from '../../MultiProvider.js';
+import {
+  TypedTransaction,
+  TypedTransactionReceipt,
+} from '../../ProviderType.js';
+
+import { TxSubmitterType } from './TxSubmitterTypes.js';
+
+export interface TxSubmitterInterface<
+  TX extends TypedTransaction,
+  TR extends TypedTransactionReceipt,
```

Comment:
you could instead just take `TProtocol extends ProtocolType` and infer the tx/receipt like https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/3a08e31b6d603a4eec691da4218a965c7c0a8283/typescript/sdk/src/crud/AbstractCrudModule.ts#L21

e.g. 
```ts
// TX
ProtocolTypedTransaction<TProtocol>

//TR
ProtocolTypedReceipt<TProtocol>
```

Reply:
np - it was only merged into main the other day